### ISSUE TITLE
false positive with `Undefined name Pool F821` in .pyi 

### ISSUE THREAD

**Issue by sbdchd** [charliermarsh/ruff#1613]
minimal reproducible example of [a larger module type stub](https://github.com/chdsbd/kodiak/blob/0c1171bc8f574d7ee11ab6174036edadfbc5ec09/bot/typings/asyncio_redis/pool.pyi#L4-L4) aka `.pyi` file:

```python
# pool.pyi
class Pool:
    @classmethod
    def create(cls) -> Pool: ...
```

errors with:

```
Undefined name `Pool`Ruff(F821)
```

```
typings/asyncio_redis/pool.pyi:31:10: F821 Undefined name `Pool`
```

**Comment by charliermarsh** [comment]
Iiiinteresting... Check out https://pypi.org/project/flake8-pyi/:

```
from __future__ import annotations has no effect in stub files, since type checkers automatically treat stubs as having those semantics.
```

I didn't realize that.
